packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -2,35 +2,139 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-
 import 'dart:async';
 import 'dart:ui' as ui show Image, Codec, FrameInfo;
 import 'dart:ui' show hashValues;
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/scheduler.dart';
 
+/// A reference to a [ui.Image] object.
+///
+/// To get an ImageHandle, clients should call [ImageInfo.obtainImageHandle].
+///
+/// Clients of an [ImageStream] should call [dispose] on this object when they
```

Comment:
This API still seems ... akward? In order to access `ImageInfo.image` I need to call a different method (`ImageInfo.obtainImageHandle`) first. Could that method just return the thing I want so it's actually natural to call it first?

Reply:
I think to do that we'd have to make this a breaking change. This is an attempt to keep this "less-breaking" (it still requires that we make ImageInfo non-const).

I toyed with the idea of returning something that `implements` Image, which when you call `dispose` on it actually just decreases a refcount. The problem with that will be it isn't usable on the native side because it's not a real native backed object.

We could break this by `@deprecated` the image getter, and exposing the `image` off of `ImageHandle`, which you then dispose. Would that be better?